p2p/agent.go

```diff
@@ -236,59 +260,93 @@ func (p *agent) Start(ctx context.Context) error {
 		return errors.Wrap(err, "error when instantiating Agent host")
 	}
 
-	if err := host.AddBroadcastPubSub(ctx, _broadcastTopic+p.topicSuffix, func(ctx context.Context, data []byte) (err error) {
-		// Blocking handling the broadcast message until the agent is started
-		<-ready
-		var (
-			peerID    string
-			broadcast iotexrpc.BroadcastMsg
-			latency   int64
-		)
-		skip := false
-		defer func() {
-			// Skip accounting if the broadcast message is not handled
-			if skip {
-				return
+	if err := host.AddBroadcastPubSub(
+		ctx,
+		_broadcastTopic+p.topicSuffix,
+		func(ctx context.Context, pid peer.ID, msg *pubsub.Message) pubsub.ValidationResult {
+			if pid.String() == host.HostIdentity() {
+				return pubsub.ValidationAccept
```

Comment:
should return `ValidationIgnore`? skip the message from itself at left-side L271

Reply:
message won't be broadcasted if ignored